src/fs-cache.js

```diff
@@ -125,35 +125,36 @@ module.exports = function(params, callback) {
   const identifier = params.identifier;
   let directory;
 
-  if (typeof params.directory === "string") {
-    directory = params.directory;
-  } else {
-    directory = findCacheDir({ name: "babel-loader" }) || os.tmpdir();
+  try {
+    if (typeof params.directory === "string") {
+      directory = params.directory;
+      mkdirp.sync(directory);
+    } else {
+      directory = findCacheDir({ name: "babel-loader", create: true });
```

Comment:
Yes, but passing `create: true` makes `findCacheDir` create the directory, which will force it to throw if it doesn't exist.

It would make sense to turn this into a test case.